### ISSUE TITLE
/usr/bin/ld: cannot find -lmdb

### ISSUE THREAD

**Issue by p4tr1ckc4rs0n** [NVIDIA/DIGITS#862]
Hi all, im getting this error when I try and make caffe. I am making it with CPU_ONLY. I really dont understand what is going on - I have an earlier build of caffe (not nv fork) and when I make it, it's fine. This is the error I get. Any help would be greatly appreciated!

AR -o .build_release/lib/libcaffe-nv.a
LD -o .build_release/lib/libcaffe-nv.so.0.11.0
/usr/bin/ld: cannot find -lmdb
collect2: error: ld returned 1 exit status

**Comment by lukeyeager** [comment]
Looks like you're missing the LMDB library. Did you install `liblmdb-dev`? Which instructions are you following?

**Comment by TimZaman** [comment]
Do as Luke said or uncomment caffe makefile line: 
`# USE_LMDB := 0`
https://github.com/BVLC/caffe/blob/master/Makefile.config.example#L13

**Comment by p4tr1ckc4rs0n** [comment]
Hi, thanks for the relpy.

These are the instructions that i followed: http://hanzratech.in/2015/07/27/installing-caffe-on-ubuntu.html

Yes i installed liblmb-dev. Running ldconfig -p | grep lmdb i get: 

	liblmdb.so.0 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/liblmdb.so.0
	liblmdb.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/liblmdb.so

I added the path to the .so file in the 'INCLUDE_DIRS :=" in the caffe makefile - still it didnt work.

I have an older build of caffe on another computer - oddly enough when i copied it across to my current computer and made caffe everything worked. 

I 'solved' the problem by making caffe via cmake instead... thanks for the help.

**Comment by TimZaman** [comment]
> I added the path to the .so file in the 'INCLUDE_DIRS :=" in the caffe makefile - still it didnt work.

The `INCLUDE_DIRS` is where the header files generally reside. You're going to want to add a library to `LIBRARY_DIRS` of course 😄 , where the libraries reside.
So in 
https://github.com/BVLC/caffe/blob/master/Makefile.config.example#L91
Change
`LIBRARY_DIRS := $(PYTHON_LIB) /usr/local/lib /usr/lib`
To
`LIBRARY_DIRS := $(PYTHON_LIB) /usr/local/lib /usr/lib /usr/lib/x86_64-linux-gnu/`

**Comment by lukeyeager** [comment]
> These are the instructions that i followed

@p4tr1ckc4rs0n When building Caffe for DIGITS, I suggest you use our documentation:
https://github.com/NVIDIA/DIGITS/blob/digits-4.0/docs/BuildCaffe.md

Glad you got it to work with CMake!

**Comment by TM0809** [comment]
Hello all, 

I am having the same problem. I changed the include directory as suggested by @TimZaman . But I am not able to get caffe to build. Please help me out. Here is the error I am getting followed by snippets of my Makefile.Config
![screenshot from 2017-02-23 00-51-48](https://cloud.githubusercontent.com/assets/25973049/23246557/8b2d3492-f962-11e6-9119-d05ce91ee9fb.png)

![screenshot from 2017-02-23 00-50-56](https://cloud.githubusercontent.com/assets/25973049/23246559/9036d92a-f962-11e6-81ef-cb13cf99f989.png)

**Comment by TM0809** [comment]
@lukeyeager  @TimZaman I am having a similar issue with llmdb files. I have included the directories you have suggested in the previous messages. I am getting the following error messages. I have also attached the Makefile.config. 

![screenshot from 2017-02-23 00-50-56](https://cloud.githubusercontent.com/assets/25973049/23246689/695a975a-f963-11e6-893a-769b1f90ebf0.png)

![screenshot from 2017-02-23 00-51-48](https://cloud.githubusercontent.com/assets/25973049/23246693/6c51714a-f963-11e6-816b-51418c2154f2.png)
